email-smtp/it/AsynchronousIntegrationTest.java

```diff
@@ -0,0 +1,83 @@
+import io.commercetools.sunrise.email.MessageEditor;
+import io.commercetools.sunrise.email.smtp.SmtpAuthEmailSender;
+import org.junit.Test;
+
+import javax.annotation.Nonnull;
+import javax.mail.internet.MimeMessage;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AsynchronousIntegrationTest extends AbstractIntegrationTest {
+
+    @Test
+    public void editorIsFinishedBeforeSendReturns() {
+        final SlowButFinishingMessageEditor editor = new SlowButFinishingMessageEditor();
+        final CompletionStage<String> completionStage = sender.send(editor);
+        final boolean editorFinished = editor.finished;
+        // Wait for sending to complete so the test is done when this method exits
+        // and does not interfere with other tests.
+        completionStage.toCompletableFuture().join();
+        assertThat(editorFinished).isTrue();
+    }
+
+    @Test
+    public void sendTwoMessagesAndWaitForBoth() {
```

Comment:
I think it is not necessary to guarantee the order of the messages. Some servers are slower than others and emails can differ in sizes so it is reasonable that a later send email still can arrive before the first.